bcftbx/cmdparse.py

```diff
@@ -234,90 +242,104 @@ def print_command(self,cmd,message=None):
+    add_cmd('--nprocessors', action='store',
+            dest='nprocessors', default=default_nprocessors,
+            help="explicitly specify number of processors/cores to use "
+            "(default %s)" %default_display)
     return parser
 
 def add_runner_option(parser):
     """Add a '--runner' option to a parser
-    
-    Given an OptionParser 'parser', add a '--runner' option.
+
+    Given an ArgumentParser 'parser', add a '--runner' option.
 
     The value of this option can be accessed via the 'runner'
     attribute of the parser options (use the 'fetch_runner'
     function to return a JobRunner object from the supplied
     value).
 
-    Returns the input OptionParser object.
+    Returns the input ArgumentParser object.
 
     """
-    parser.add_option('--runner',action='store',
-                      dest='runner',default=None,
-                      help="explicitly specify runner definition (e.g. "
-                      "'GEJobRunner(-j y)')")
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
+    add_cmd('--runner', action='store',
+            dest='runner', default=None,
+            help="explicitly specify runner definition (e.g. "
+            "'GEJobRunner(-j y)')")
     return parser
 
 def add_no_save_option(parser):
     """Add a '--no-save' option to a parser
-    
-    Given an OptionParser 'parser', add a '--no-save' option.
+
+    Given an ArgumentParser 'parser', add a '--no-save' option.
 
     The value of this option can be accessed via the 'no_save'
     attribute of the parser options.
 
-    Returns the input OptionParser object.
+    Returns the input ArgumentParser object.
 
     """
-    parser.add_option('--no-save',action='store_true',dest='no_save',default=False,
-                      help="Don't save parameter changes to the auto_process.info file")
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
+    add_cmd('--no-save', action='store_true', dest='no_save', default=False,
+            help="Don't save parameter changes to the auto_process.info file")
     return parser
 
 def add_dry_run_option(parser):
     """Add a '--dry-run' option to a parser
-    
-    Given an OptionParser 'parser', add a '--dry-run' option.
+
+    Given an ArgumentParser 'parser', add a '--dry-run' option.
 
     The value of this option can be accessed via the 'dry_run'
     attribute of the parser options.
 
-    Returns the input OptionParser object.
-    
+    Returns the input ArgumentParser object.
+
     """
-    parser.add_option('--dry-run',action='store_true',dest='dry_run',default=False,
-                      help="Dry run i.e. report what would be done but don't perform "
-                      "any actions")
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
```

Comment:
It's preferable to split the clause following the `else` statement onto a separate new line.